Content.Server/GameObjects/Components/Movement/PullableComponent.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using Content.Server.GameObjects.EntitySystems;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Movement
+{
+    [RegisterComponent]
+    public class PullableComponent: Component
+    {
+        private const float DistBeforePull = 1.0f;
+
+        private const float DistBeforeStopPull = InteractionSystem.InteractionRange;
+
+        private PhysicsComponent _physicsComponent;
+
+        private IEntity _puller;
+        public override string Name => "Pullable";
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public IEntity Puller => _puller;
+
+        public bool GettingPulled => _puller != null;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _physicsComponent = Owner.GetComponent<PhysicsComponent>();
+
+            _puller = null;
+        }
+
+        public void GetPulled(IEntity puller)
```

Comment:
I would have called this `StarPull` to make it clear it is not a getter, and to make the parallel with `StopPull`